src/sentry/api/endpoints/organization_unsubscribe.py

```diff
@@ -0,0 +1,124 @@
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.authentication import SignedRequestAuthentication
+from sentry.api.base import Endpoint, region_silo_endpoint
+from sentry.db.models.base import BaseModel
+from sentry.models.group import Group
+from sentry.models.groupsubscription import GroupSubscription
+from sentry.models.organizationmember import OrganizationMember
+from sentry.models.project import Project
+from sentry.notifications.types import NotificationSettingOptionValues, NotificationSettingTypes
+from sentry.services.hybrid_cloud.actor import ActorType, RpcActor
+from sentry.services.hybrid_cloud.notifications.service import notifications_service
+from sentry.types.integrations import ExternalProviders
+
+T = TypeVar("T", bound=BaseModel)
+
+
+class OrganizationUnsubscribeBase(Endpoint, Generic[T]):
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+        "POST": ApiPublishStatus.PRIVATE,
+    }
+    owner = ApiOwner.HYBRID_CLOUD
+    authentication_classes = (SignedRequestAuthentication,)
+    permission_classes = ()
+
+    object_type = "unknown"
+
+    def fetch_instance(self, request: Request, organization_slug: str, id: int) -> T:
+        raise NotImplementedError()
+
+    def unsubscribe(self, request: Request, instance: T):
+        raise NotImplementedError()
+
+    def add_instance_data(self, data: dict[str, Any], instance: T) -> dict[str, Any]:
+        return data
+
+    def get(self, request: Request, organization_slug: str, id: int, **kwargs) -> Response:
+        if not request.user_from_signed_request:
+            raise NotFound()
+        instance = self.fetch_instance(request, organization_slug, id)
+        view_url = ""
+        if hasattr(instance, "get_absolute_url"):
+            view_url = str(instance.get_absolute_url())
+        data = {
+            "viewUrl": view_url,
+            "type": self.object_type,
+        }
+        return Response(self.add_instance_data(data, instance), 200)
+
+    def post(self, request: Request, organization_slug: str, id: int, **kwargs) -> Response:
+        if not request.user_from_signed_request:
+            raise NotFound()
+        instance = self.fetch_instance(request, organization_slug, id)
+
+        if request.data.get("cancel"):
+            self.unsubscribe(request, instance)
+        return Response(status=201)
+
+
+@region_silo_endpoint
+class OrganizationUnsubscribeProject(OrganizationUnsubscribeBase[Project]):
+    object_type = "project"
+
+    def fetch_instance(self, request: Request, organization_slug: str, id: int) -> Project:
+        try:
+            project = Project.objects.select_related("organization").get(id=id)
+        except Project.DoesNotExist:
+            raise NotFound()
+        if project.organization.slug != organization_slug:
+            raise NotFound()
```

Comment:
Is there a reason to not do that org slug check in the filter itself? Doesn't need to change, just curious

ex:
```python
project = Project.objects.select_related("organization").get(id=id, organization__slug=organization_slug)
```

Reply:
Oh, I see - this better matches the `Group` behavior below